drivers/media/i2c/arducam_64mp.c

```diff
@@ -2291,6 +2297,19 @@ static int arducam_64mp_init_controls(struct arducam_64mp *arducam_64mp)
 						     ARDUCAM_64MP_PIXEL_RATE,
 						     ARDUCAM_64MP_PIXEL_RATE, 1,
 						     ARDUCAM_64MP_PIXEL_RATE);
+	if (arducam_64mp->pixel_rate)
+		arducam_64mp->pixel_rate->flags |= V4L2_CTRL_FLAG_READ_ONLY;
+
+	/* LINK_FREQ is also read only */
+	link_freq_max = ARRAY_SIZE(arducam_64mp_link_freq_menu) - 1;
+	arducam_64mp->link_freq =
+		v4l2_ctrl_new_int_menu(ctrl_hdlr, &arducam_64mp_ctrl_ops,
+				       V4L2_CID_LINK_FREQ,
+				       link_freq_max,
+				       0,
+				       arducam_64mp_link_freq_menu);
+	if (arducam_64mp->link_freq)
+		arducam_64mp->link_freq->flags |= V4L2_CTRL_FLAG_READ_ONLY;
```

Comment:
If you're only needing the `struct v4l2_ctrl` for setting the READ_ONLY flag, then store it as a local to this function rather than in the sensor's state.

Reply:
Done.